doc/user/content/sql/create-source/kafka.md

```diff
@@ -0,0 +1,371 @@
+
+Field                                   | Value  | Description
+----------------------------------------|--------|----------------------------------------
+`security_protocol`                     | `text` | Use `plaintext`, `ssl`, `sasl_plaintext` or `sasl_ssl` to connect to the Kafka cluster.
+`sasl_mechanisms`                       | `text` | The SASL mechanism to use for authentication. Supported: `GSSAPI` (the default), `PLAIN`, `SCRAM-SHA-256`, `SCRAM-SHA-512`.
+`sasl_username`                         | `text` | Your SASL username, if any. Required if `sasl_mechanisms` is `PLAIN`.
+`sasl_password`                         | `text` | Your SASL password, if any. Required if `sasl_mechanisms` is `PLAIN`.<br/><br/>This option stores the password in Materialize's on-disk catalog. For an alternative, use `sasl_password_env`.
+`sasl_password_env`                     | `text` | Use the value stored in the named environment variable as the value for `sasl_password`. <br/><br/>This option does not store the password on-disk in Materialize's catalog, but requires the environment variable's presence to boot Materialize.
+`sasl_kerberos_keytab`                  | `text` | The absolute path to your keytab. Required if `sasl_mechanisms` is `GSSAPI`.
+`sasl_kerberos_kinit_cmd`               | `text` | Shell command to refresh or acquire the client's Kerberos ticket. Required if `sasl_mechanisms` is `GSSAPI`.
+`sasl_kerberos_min_time_before_relogin` | `text` | Minimum time in milliseconds between key refresh attempts. Disable automatic key refresh by setting this property to 0. Required if `sasl_mechanisms` is `GSSAPI`.
+`sasl_kerberos_principal`               | `text` | Materialize Kerberos principal name. Required if `sasl_mechanisms` is `GSSAPI`.
+`sasl_kerberos_service_name`            | `text` | Kafka's service name on its host, i.e. the service principal name not including `/hostname@REALM`. Required if `sasl_mechanisms` is `GSSAPI`.
+
+## Examples
+
+### Creating a source
+
+{{< tabs tabID="1" >}}
+{{< tab "Avro">}}
+
+```sql
+CREATE SOURCE avro_source
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'data'
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'https://localhost:8081';
+```
+
+{{< /tab >}}
+{{< tab "JSON">}}
+
+```sql
+CREATE SOURCE json_source
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'data'
+  FORMAT BYTES;
+```
+
+```sql
+CREATE MATERIALIZED VIEW jsonified_kafka_source AS
+  SELECT
+    data->>'field1' AS field_1,
+    data->>'field2' AS field_2,
+    data->>'field3' AS field_3
+  FROM (SELECT text::jsonb AS data FROM json_source);
+```
+
+{{< /tab >}}
+{{< tab "Protobuf">}}
+
+```sql
+CREATE SOURCE proto_source
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'billing'
+  WITH (cache = true)
+  FORMAT PROTOBUF MESSAGE '.billing.Batch'
+  USING SCHEMA FILE '[path to schema]';
+```
+
+{{< /tab >}}
+{{< tab "Text/bytes">}}
+
+```sql
+CREATE SOURCE text_source
+  FROM KAFKA BROKER 'localhost:9092' TOPIC 'data'
+  FORMAT TEXT
```

Comment:
It bothers me that we call this `Text/bytes` but tell people to use `FORMAT TEXT`, while for JSON we tell people to use `FORMAT BYTES` and decode the JSON directly. It would be better to properly specify somewhere what do we mean by `TEXT` (I guess this is one of the separate pages?) and be explicit that `FORMAT TEXT` and `FORMAT BYTES` are two different things.